src/commands/build/variants.ts

```diff
@@ -0,0 +1,71 @@
+import chalk from "chalk";
+import { getAllVariantsInPath } from "../../files/variants/getAllPackageVariants.js";
+
+/**
+ * Main function to retrieve the valid variant names based on the specified variants and available directories.
+ */
+export function getVariantNames({
+  variantsDirPath,
+  variants
+}: {
+  variantsDirPath: string;
+  variants?: string;
+}): string[] {
+  const allVariantNames = getAllVariantsInPath(variantsDirPath);
```

Comment:
Are directories all thats needed to define valid variant names for a package? 

I think that ideally, the manifest of a package should contain the valid variants of a package, not its directory tree

Reply:
Doing it this way there is no need to input the variant names anywhere else than in the package variants directory. Actually, it is not necessary to define them in the manifest because it would not make sense to include other directories inside the package variants dir rather than the variants themselves